### DISCUSSION TITLE
Open Source Quotes

### DISCUSSION THREAD

**Discussion by naomi-lgbt** [Heptagram-Project/api#25] [General]
When a quote doesn't have author data, the response includes a request to email you if someone knows the author. Rather than an email based approach, why not open source the quotes?

The structure of the data is simple enough to be effective in a JSON object, can be hosted directly in the API files, and then you'd be able to accept contributions.

Could also consider this for your jokes data as well.

**Comment by jaspermayone** [comment]
We did do this to start, but as the database of various things grows, it makes more sense to have it in something like a MongoDB Atlas instance. Wrather than migrating **everything** when it gets to that point. I would rather have them already migrated as much as possible. Currently there is about 100 quotes that need authors, and I have that data with authors stored, I just need to finish updating it. The /add endpoints all have validation on them, meaning that there won't be any more data lacking authors.

**Comment by jaspermayone** [comment]
Controlling the data like this also allows us to make sure no inappropriate jokes or quotes are added to the dataset.

**Comment by jaspermayone** [comment]
I am going to keep this open as an discussion, in case others want to chime in.

**Comment by naomi-lgbt** [comment]
To be fair, freeCodeCamp does this approach with a [large number of quotes](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/i18n/locales/english/motivation.json).

I also take this approach in [my own project](https://github.com/nhcarrigan/inspiration-bot/blob/main/src/config/QuoteList.ts) which even allows me to [test for duplicates](https://github.com/nhcarrigan/inspiration-bot/blob/main/test/quotes.spec.ts).
